### PR TITLE
Add LeakCanary +  Extract syncing of data from SyncWorker into DataSync

### PR DESCRIPTION
**Pull request by saket**

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/sync/DataSync.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import org.simple.clinic.patient.sync.PatientSync
+import org.simple.clinic.protocol.sync.ProtocolSync
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.ErrorResolver
+import org.simple.clinic.util.ResolvedError
+import org.simple.clinic.util.exhaustive
+import timber.log.Timber
+import javax.inject.Inject
+
+class DataSync @Inject constructor(
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/sync/DataSync.kt]: Should this be `@AppScope` scoped?

Comment (saket) [reply app/src/main/java/org/simple/clinic/sync/DataSync.kt]: It's not maintaining any mutable state, so I don't see a reason for retaining this in memory throughout the app's life.

Comment (saket) [reply app/src/main/java/org/simple/clinic/sync/DataSync.kt]: Hrm, I don't think so. It's not maintaining any mutable state, so I don't see a reason for retaining this in memory throughout the app's life.

Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/sync/DataSync.kt]: 👍

**Pull request merged**